app/views/unattended/provisioning_templates/snippet/subscription_manager_setup.erb

```diff
@@ -0,0 +1,140 @@
+<%#
+kind: snippet
+name: subscription_manager_setup
+model: ProvisioningTemplate
+snippet: true
+description: |
+  Streamline the provisioning process by configuring subscription-manager during the Global Registration template execution.
+  This snippet integrates into the existing "redhat_register" snippet for efficient host registration.
+-%>
+
+if [ -z "$PKG_MANAGER" ]; then
+ <%= snippet 'pkg_manager' -%>
+fi
```

Comment:
To resolve https://github.com/theforeman/foreman/pull/9837#discussion_r1369755906